File: csharp_solutions/two_sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int> numTable = new Dictionary<int,int>();
        for (int i=0; i<nums.Length; i++) {
            int neededNum = target - nums[i];
            if (numTable.ContainsKey(neededNum)) {
                return new int[]{i, numTable[neededNum]};
            }
            else {
                numTable[nums[i]] = i;
            }
        }
        return new int[]{-1,-1};
    }
}

File: python_solutions/two_sum.py
class Solution:
    def twoSum(self, nums, target: int):
        nums_dict = {}
        for i in range(len(nums)):
            nums_dict[nums[i]] = i
        for i in range(len(nums)):
            looking_for = target - nums[i]
            if looking_for in nums_dict and nums_dict[looking_for] != i:
                return [i, nums_dict[looking_for]]
        